mage_ai/io/file.py

```diff
@@ -38,10 +38,15 @@ def export(self, df: DataFrame, **kwargs) -> None:
             self._write(df, self.filepath, **kwargs)
 
     @classmethod
-    def with_config(cls, config: Mapping[str, Any]) -> 'FileIO':
-        try:
-            return cls(**config[IOConfigKeys.FILE])
-        except KeyError:
-            raise KeyError(
-                f'No configuration settings found for \'{IOConfigKeys.FILE}\' under profile'
-            )
+    def with_config(
```

Comment:
When are we going to use this method? The type of config is also different here.

Reply:
Ah I think I messed a couple of things up with some of the interface design here. The problem is that this PR was initially a bit larger but I stripped it down after the fact. I'll have to refactor some of the abstract classes so that `File` doesn't require a `with_config` function.